lib/rules/selector-max-specificity/index.mjs

```diff
@@ -80,148 +57,46 @@ const rule = (primary, secondaryOptions) => {
-			} else if (aNPlusBOfSNotationPseudoClasses.has(ownValue.replace(/^:/, ''))) {
-				return selectorSpecificity(ignoreOfSelectorIfAny(node));
-			} else {
-				ownSpecificity = selectorSpecificity(node.clone({ nodes: [] }));
-			}
-
-			return specificitySum([ownSpecificity, maxChildSpecificity(node)]);
-		};
-
-		/**
-		 * @param {import('postcss-selector-parser').Node} node
-		 * @returns {boolean}
-		 */
-		const shouldSkipPseudoClassArgument = (node) => {
-			// postcss-selector-parser includes the arguments to nth-child() functions
-			// as "tags", so we need to ignore them ourselves.
-			// The fake-tag's "parent" is actually a selector node, whose parent
-			// should be the :nth-child pseudo node.
-			const parentNode = node.parent && node.parent.parent;
-
-			if (parentNode && parentNode.type === 'pseudo' && parentNode.value) {
-				const pseudoClass = parentNode.value.toLowerCase().replace(/^:/, '');
-
-				return (
-					aNPlusBNotationPseudoClasses.has(pseudoClass) || linguisticPseudoClasses.has(pseudoClass)
-				);
-			}
-
-			return false;
-		};
-
-		/**
-		 * Calculate the specificity of a node parsed by `postcss-selector-parser`.
-		 *
-		 * @param {import('postcss-selector-parser').Node} node
-		 * @returns {Specificity}
-		 */
-		const nodeSpecificity = (node) => {
-			if (shouldSkipPseudoClassArgument(node)) {
-				return zeroSpecificity();
-			}
-
-			switch (node.type) {
-				case 'attribute':
-				case 'class':
-				case 'id':
-				case 'tag':
-					return simpleSpecificity(node);
-				case 'pseudo':
-					return pseudoSpecificity(node);
-				case 'selector':
-					// Calculate the sum of all the direct children
-					return specificitySum(node.map((n) => nodeSpecificity(n)));
-				default:
-					return zeroSpecificity();
-			}
-		};
+		/** @type {import('@csstools/selector-specificity').CustomSpecificityCallback | undefined} */
+		const customSpecificity = secondaryOptions?.ignoreSelectors
+			? (node) => {
+					switch (node.type) {
+						case 'attribute':
+						case 'class':
+						case 'id':
+						case 'tag':
+							if (!isSelectorIgnored(node.toString())) {
+								return;
+							}
+
+							return zeroSpecificity();
+						case 'pseudo': {
+							if (!isSelectorIgnored(node.value.toLowerCase())) {
+								return;
+							}
+
+							if (!node.nodes?.length) {
+								return zeroSpecificity();
+							}
+
+							// We only ignore the current pseudo-class, not the specificity of the child nodes.
+							// Calculate the diff between specificity with and without child nodes.
+							const entireSpecificity = selectorSpecificity(node);
+
+							node.nodes = [];
+							const emptySpecificity = selectorSpecificity(node);
+
+							return {
+								a: entireSpecificity.a - emptySpecificity.a,
+								b: entireSpecificity.b - emptySpecificity.b,
+								c: entireSpecificity.c - emptySpecificity.c,
+							};
+						}
+						default:
+						// Other node types are not ignorable.
```

Comment:
[suggest] Super-minor, you can ignore it. If adding the `return` statement, Prettier would align indent:

```suggestion
							// Other node types are not ignorable.
							return;
```

Reply:
Eslint errors on this suggestion with `Unnecessary return statement`, so I've reverted this back to what it was.

But I agree that the indent should be as suggested :)